### PR TITLE
Project file removal

### PR DESCRIPTION
**Pull request by stefanceriu**
With githooks that setup the full development environment now in place we can assume that everybody has XcodeGen installed so we can go ahead and remove the project file from source control. This will save us a lot of trouble with always having to fix conflicts on it.

This PR does that and also takes the opportunity to refactor how we handle dependencies on the CI. There are new scripts under `ci_script/ci_common` that setup the environment for both Xcode Cloud and Github Actions. 

It also removes the `brewfile` and `requirements.txt`, moves the `validate_lfs` workflow to the pre-commit hook and fixes date formatting unit tests that were failing during the weekend.

N.B. We have tried converging on these scripts withing the Tools Swift package but that fails on Github 

With `Yams` (thought that dependency was the problem but no)

```
dyld[1135]: Symbol not found: (_$s10Foundation10NSNotFoundSivg)
  Referenced from: '/Users/runner/work/element-x-ios/element-x-ios/.build/x86_64-apple-macosx/debug/tools'
  Expected in: '/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation'
/Users/runner/work/_temp/b[87](https://github.com/vector-im/element-x-ios/actions/runs/4146224225/jobs/7171665941#step:4:88)51cf0-502b-46da-a555-3d0dadac454b.sh: line 1:  1135 Abort trap: 6
```

without `Yams`

```
dyld[1128]: Symbol not found: (_$s10Foundation11JSONEncoderC16OutputFormattingV10sortedKeysAEvgZ)
  Referenced from: '/Users/runner/work/element-x-ios/element-x-ios/.build/x86_64-apple-macosx/debug/tools'
  Expected in: '/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation'
```

### CODE REVIEW

File: ci_scripts/ci_common.sh [hunk 1]
```diff
@@ -29,11 +29,25 @@ setup_environment () {
...
-install_python_dependencies () {
-    pip3 install -r requirements.txt # Install towncrier for generating changelogs
+install_xcode_cloud_python_dependencies () {
+    pip3 install towncrier # Install towncrier for generating changelogs
+}
+
+setup_github_actions_environment() {
+    brew install "xcodegen"
+    brew install "swiftformat"
+    brew install "git-lfs"
```
Comment (pixlwave) [inline ci_scripts/ci_common.sh:44]: Seems cleaner to run this in a single pass?
```suggestion
    brew install xcodegen swiftformat git-lfs
```

**Review by pixlwave — APPROVED** [review]
Overall looks good to me, but I'm not convinced by the changes to the date formatting tests.

File: Brewfile [hunk 2]
```diff
@@ -1,7 +0,0 @@
-brew "xcodegen"
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply Brewfile:1]: Bah, I was meaning to move the brew stuff directly to there but I forgot

File: UnitTests/Sources/DateTests.swift [hunk 3]
```diff
@@ -24,20 +24,20 @@ class DateTests: XCTestCase {
     let startOfYesterday = Calendar.current.startOfDay(for: Calendar.current.date(byAdding: .day, value: -1, to: .now)!)
     
     func testMinimalDateFormatting() {
-        let today = calendar.date(byAdding: DateComponents(hour: 9, minute: 30), to: startOfToday)
-        XCTAssertEqual(today?.formattedMinimal(), "9:30 AM")
+        let today = calendar.date(byAdding: DateComponents(hour: 9, minute: 30), to: startOfToday)!
+        XCTAssertEqual(today.formattedMinimal(), today.formatted(date: .omitted, time: .shortened))
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply UnitTests/Sources/DateTests.swift:28]: Well they don't use the same logic, just the same formatting. What we are checking here is that dates get formatted accordingly based on how far back they are from present

File: .github/workflows/validate-lfs.yml [hunk 4]
```diff
@@ -1,19 +0,0 @@
-name: Validate Git LFS
```
[the remark this answers is not in the document]
Comment (pixlwave) [reply .github/workflows/validate-lfs.yml:1]: Ohhhh, ok makes sense now. And I guess if it does go wrong, someone on the team will detect it locally anyway :)

[the remark this answers is not in the document]
Comment (stefanceriu) [reply .github/workflows/validate-lfs.yml:1]: Yep, that's what I'm thinking

**Pull request merged**